internal/net/grpc/pool.go

```diff
@@ -0,0 +1,213 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package grpc provides generic functionality for grpc
+package grpc
+
+import (
+	"context"
+	"net"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/log"
+	"google.golang.org/grpc"
+)
+
+type ClientConnPool struct {
+	ctx     context.Context
+	conn    *ClientConn // default connection
+	pool    sync.Pool
+	addr    string
+	host    string
+	port    string
+	size    uint64
+	length  uint64
+	dopts   []DialOption
+	closing atomic.Value
+}
+
+func NewPool(ctx context.Context, addr string, size uint64, dopts ...DialOption) (*ClientConnPool, error) {
+	if !strings.HasPrefix(addr, "::") && strings.HasPrefix(addr, ":") {
+		addr = "127.0.0.1" + addr
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
"127.0.0.1" appears more than three times in this file. it should be const, i think.

Reply:
fixed!